plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -499,6 +509,36 @@ public MongoCursor<Document> execute(MongoTableHandle tableHandle, List<MongoCol
         return iterable.iterator();
     }
 
+    /**
+     * Creates a set of sufficient columns for the input projected columns. For example,
+     * if input {@param columns} include columns "a.b" and "a.b.c", then they will be projected from a single column "a.b".
+     */
+    public static List<MongoColumnHandle> getSufficientColumns(List<MongoColumnHandle> columnHandles)
+    {
+        List<MongoColumnHandle> sufficientColumns = new ArrayList<>();
+        for (MongoColumnHandle column : columnHandles) {
+            String qualifiedName = column.getQualifiedName();
+            List<MongoColumnHandle> columnsToRemove = new ArrayList<>();
+            boolean isAddable = true;
+
+            for (MongoColumnHandle existingColumn : sufficientColumns) {
+                String name = existingColumn.getQualifiedName();
+                if (name.startsWith(qualifiedName + ".")) {
```

Comment:
This looks a bit brittle - is it possible to use the `dereferenceNames` to validate if a parent column bas been referred or not ?